src/posix/FileSystemPosix.cpp

```diff
@@ -57,7 +57,20 @@ namespace FileSystem {
 #ifdef __APPLE__
 		path += "Library/Application Support/Pioneer";
 #else
-		path += ".pioneer";
+		struct stat info;
+		stat((path + ".pioneer").c_str(), &info);
+		if (S_ISDIR(info.st_mode)) {
+			// Check for legacy pioneer directory.
+			path += ".pioneer";
+		} else {
+			char *state_home = getenv("XDG_DATA_HOME");
+			if (state_home != NULL) {
+				path += state_home;
+				path += "/pioneer";
```

Comment:
According to what I can find, `XDG_DATA_HOME` is intended to be an absolute path, defaulting to `$HOME/.local/share`. This form would result in `$HOME/$HOME/.local/share` which would result in unintended directory nesting at best and a malformed path at worst.

Additionally, this code does not handle the (possible, albeit improbable) case where XDG_DATA_HOME is set to an empty string (e.g. by a user-specific config) and `getenv()` returns a valid string pointer with length 0.

Other than this issue, I am happy with this change!